src/nft/components/explore/Banner.tsx

```diff
@@ -0,0 +1,195 @@
+                </div>
+              </CollectionWrapper>
+            </Box>
+          </Link>
+        ))
+      ) : (
+        <>
+          {/* TODO: Improve Loading State */}
+          <p>Loading</p>
+        </>
+      )}
+    </Box>
+  )
+}
+
+export default Banner
+
+/* Collection Wrapper: applies background image to entire banner */
+const CollectionWrapper = ({ bannerImageUrl, children }: { bannerImageUrl: string; children: ReactNode }) => (
+  <div className={styles.bannerWrap} style={{ backgroundImage: `url(${bannerImageUrl})` }}>
+    <Box className={styles.bannerOverlay} width="full" />
+    {children}
+  </div>
+)
+
+/* Collection Details: displays collection stats within Banner  */
+const CollectionDetails = ({
+  collection,
+  rank,
+  hovered,
+}: {
+  collection: TrendingCollection
+  rank: number
+  hovered: boolean
+}) => (
+  <Column className={styles.collectionDetails} paddingTop="16">
+    <div>
+      <span className={styles.volumeRank}>#{rank} volume in 24hr</span>
+    </div>
+    <Row>
+      <Box as="span" marginTop="16" className={clsx(header1, styles.collectionName)}>
+        {collection.name}
+      </Box>
+      {collection.isVerified && (
+        <Box as="span" marginTop="24">
+          <VerifiedIcon height="32" width="32" />
+        </Box>
+      )}
+    </Row>
+    <Row className={bodySmall} marginTop="12" color="explicitWhite">
+      <Box>
+        <Box as="span" color="darkGray" marginRight="4">
+          Floor:
+        </Box>
+        {collection.floor ? formatEthPrice(collection.floor.toString()) : '--'} ETH
+      </Box>
+      <Box>
+        {collection.floorChange ? (
+          <Box as="span" color={collection.floorChange > 0 ? 'green200' : 'error'} marginLeft="4">
+            {collection.floorChange > 0 && '+'}
+            {formatChange(collection.floorChange)}%
+          </Box>
+        ) : null}
+      </Box>
+      <Box marginLeft="24" color="explicitWhite">
+        <Box as="span" color="darkGray" marginRight="4">
+          Volume:
+        </Box>
+        {collection.volume ? putCommas(+toSignificant(collection.volume.toString())) : '--'} ETH
+      </Box>
+      <Box>
+        {collection.volumeChange ? (
+          <Box as="span" color={collection.volumeChange > 0 ? 'green200' : 'error'} marginLeft="4">
+            {collection.volumeChange > 0 && '+'}
+            {formatChange(collection.volumeChange)}%
+          </Box>
+        ) : null}
+      </Box>
+    </Row>
+    <Link
+      className={clsx(buttonMedium, styles.exploreCollection)}
+      to={`/nfts/collection/${collection.address}`}
+      style={{ textDecoration: 'none', backgroundColor: `${hovered ? vars.color.blue400 : vars.color.grey700}` }}
+    >
+      Explore collection
+    </Link>
+  </Column>
+)
+
+/* Carousel Progress indicators */
+const CarouselProgress = ({
+  length,
+  currentIndex,
+  onClick,
+}: {
+  length: number
+  currentIndex: number
+  onClick(index: number): void
```

Comment:
This is confusing, because it's a built-in event being overridden. Could you retain it with no signature, and bind the index when you render, so it's more clear what is happening?

Reply:
I refactored the carousel progress indicators a bit so hopefully the behavior is more clear